kernel/src/sched.rs

```diff
@@ -284,6 +284,14 @@ impl Kernel {
         }
     }
 
+    /// Returns whether the kernel is configured to collect debugging information for processes.
+    ///
+    /// Clients that rely on this information, e.g. process console, should call this method
+    /// to check if process debugging is enabled and warn the user if it is not.
+    pub fn process_debugging_enabled() -> bool {
```

Comment:
Rather than this, which is very easy to forget to call, what about making the debug functions that can return useless values return `None`?

Reply:
I added this to be able to print the warning banner in the process console.

`debug_*` functions currently return `0` if `Process::debug` is not there when `debug_processes` is disabled.  It looks like returning `None` instead of `0` would result in a lot of changes because callers expect `usize` from these functions and it could still possibly be misinterpreted by someone not familiar with the output. Please let me know what you think.

Reply:
If the kernel config needs to be exposed outside of the kernel crate, shouldn't we make the config module public, or expose a function returning a (read-only) reference to `config::CONFIG`?

Reply:
Is it safe to assume that anything in config::CONFIG should be public to capsules? 

Though I agree that a function returning a specific configuration feature probably belongs in `config.rs`, not `sched.rs`. I suppose that an alternative would be to make `Config` a pub struct, but then internally use the `crate` modifier on any fields we don't want exposed to capsules?

Reply:
@gendx, I agree but I just wasn't sure how much of it should be exposed so I exposed the new flag only. The function is implemented in `kernel/src/sched.rs` because I thought `Kernel` was a reasonable home for a kernel config but if it is OK to make `Config` a pub struct, I can definitely implement @hudson-ayers's suggestion.

Reply:
I think @hudson-ayers's suggestion would be the way to go. By the way, we should use `pub(crate)` now, as was finally decided in https://github.com/tock/tock/pull/1650#issuecomment-629399631.

On the other hand, it seems to me that instead of exposing the config to capsules, we could just add a function to the kernel that contains the following code, and only expose this function for capsules.

https://github.com/tock/tock/blob/44875ac55169c99bcb15222e2c84d0726dce3d2f/capsules/src/process_console.rs#L252-L255

The code and text for this banner is already duplicated anyway, so having a single function would also reduce the size of the binary.

https://github.com/tock/tock/blob/44875ac55169c99bcb15222e2c84d0726dce3d2f/kernel/src/process.rs#L1412-L1420

Reply:
I do think it would require at least some changes, but if a `debug_` call can now fail when before it couldn't, I think that should be reflected in its return type. Expecting callers to first check a flag in a completely different file before knowing if a return value is valid seems like an API we don't want.